File: RepoTransBench/source_projects/Python/NaturalHistoryMuseum_pyzbar/public_tests/test_zbar_library_public.py
import unittest
import sys

from pyzbar import zbar_library

class TestZBarLibraryPublic(unittest.TestCase):
    def test_lib_found(self):
        libname = zbar_library.load()[0]
        self.assertTrue(libname)
        self.assertIsInstance(libname, str)

    def test_lib_endswith_platform(self):
        libname = zbar_library.load()[0]
        # Instead of .so or .dll, just check it ends with .so/.dll/.dylib (broader)
        ends = ('.so', '.dll', '.dylib')
        self.assertTrue(libname.endswith(ends))

    def test_search_paths_include_library(self):
        # Check that at least one path returned by search_paths contains "zbar"
        results = zbar_library.search_paths()
        found = [path for path in results if "zbar" in path]
        self.assertTrue(len(found) > 0)

if __name__ == "__main__":
    unittest.main()

File: RepoTransBench/target_projects/Python/Java/NaturalHistoryMuseum_pyzbar/src/test/java/com/nhm/pyzbar/public_tests/TestZBarLibraryPublic.java
package com.nhm.pyzbar.public_tests;

import com.nhm.pyzbar.zbar_library.ZBarLibrary;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class TestZBarLibraryPublic {
    @Test
    void testLibFound() {
        Object[] result = ZBarLibrary.load();
        String libname = (String) result[0];
        assertNotNull(libname);
        assertTrue(libname instanceof String);
    }

    @Test
    void testLibEndsWithPlatform() {
        Object[] result = ZBarLibrary.load();
        String libname = (String) result[0];
        assertTrue(libname.endsWith(".so") || libname.endsWith(".dll") || libname.endsWith(".dylib"));
    }

    @Test
    void testSearchPathsIncludeLibrary() {
        java.util.List<String> results = ZBarLibrary.searchPaths();
        boolean found = results.stream().anyMatch(path -> path.contains("zbar"));
        assertTrue(found);
    }
}